app/services/analytics_events.rb

```diff
@@ -0,0 +1,55 @@
+module AnalyticsEvents
+  # @identity.idp.event_name Account Reset
+  # @param [Boolean] success
+  # @param ["cancel", "delete", "cancel token validation", "granted token validation", :notifications] event
+  # @param [String] message_id Request ID from AWS Pinpoint API
+  # @param [String] request_id Request ID from AWS Pinpoint API
+  # @param [Boolean] sms_phone
+  # @param [Boolean] totp does the user have an authentication app as a 2FA option?
+  # @param [Boolean] piv_cac does the user have PIV/CAC as a 2FA option?
+  # @param [Integer] count number of notifications sent
+  # @param [Hash] errors
+  # @param [String] user_id
+  # @param [Integer] account_age_in_days
+  # @param [Hash] mfa_method_counts
+  # Tracks events related to a user requesting to delete their account during the sign in process
+  # (because they have no other means to sign in).
+  def account_reset(
+    success:,
+    event: nil,
+    message_id: nil,
+    piv_cac: nil,
+    request_id: nil,
+    sms_phone: nil,
+    totp: nil,
+    count: nil,
+    errors: nil,
+    user_id: nil,
+    account_age_in_days: nil,
+    mfa_method_counts: nil,
+    pii_like_keypaths: nil,
+    error_details: nil,
+    email_addresses: nil
+  )
+    track_event(
+      'Account Reset',
+      {
+        success: success,
```

Comment:
One of the cons of this approach is each key needs to be written 3 times in this file (YARD, param, and then when it's passed over to `track_event`), which is very error-prone

Thanks to https://github.com/18F/identity-idp/pull/6012, we can be sure that if we forget to pass a param, we will error because it's unused